lib/blocktree/blocktree_test.go

```diff
@@ -371,7 +371,82 @@ func TestBlockTree_IsDecendantOf(t *testing.T) {
 	require.False(t, isDescendant)
 }
 
-func TestBlockTree_HighestCommonAncestor(t *testing.T) {
+func TestNodeLowestCommonAncestor(t *testing.T) {
+	root := &node{
+		hash:   common.Hash{0},
+		number: 0,
+	}
+
+	children := []*node{
+		{
+			hash:      common.Hash{1},
+			parent:    root,
+			isPrimary: true,
+			number:    1,
+		},
+		{
+			hash:      common.Hash{2},
+			parent:    root,
+			isPrimary: false,
+			number:    1,
+		},
+	}
+
+	childrenChildren := []*node{
+		{
+			hash:      common.Hash{3},
+			parent:    children[0],
+			isPrimary: true,
+			number:    2,
+		},
+		{
+			hash:      common.Hash{4},
+			parent:    children[1],
+			isPrimary: false,
+			number:    2,
+		},
+	}
+	finalChild := []*node{
+		{
+			hash:      common.Hash{5},
+			parent:    childrenChildren[1],
+			isPrimary: true,
+			number:    3,
+		},
+	}
+
+	ancestor, err := lowestCommonAncestor(children[1], root)
+	require.NoError(t, err)
+	require.NotNil(t, ancestor)
+	require.Equal(t, root.hash, ancestor)
+
+	ancestor, err = lowestCommonAncestor(children[1], children[1])
+	require.NoError(t, err)
+	require.NotNil(t, ancestor)
+	require.Equal(t, children[1].hash, ancestor)
+
+	ancestor, err = lowestCommonAncestor(children[0], children[1])
+	require.NoError(t, err)
+	require.NotNil(t, ancestor)
+	require.Equal(t, root.hash, ancestor)
+
+	ancestor, err = lowestCommonAncestor(children[0], childrenChildren[0])
+	require.NoError(t, err)
+	require.NotNil(t, ancestor)
+	require.Equal(t, children[0].hash, ancestor)
+
+	ancestor, err = lowestCommonAncestor(root, childrenChildren[0])
+	require.NoError(t, err)
+	require.NotNil(t, ancestor)
+	require.Equal(t, root.hash, ancestor)
+
+	ancestor, err = lowestCommonAncestor(finalChild[0], childrenChildren[0])
+	require.NoError(t, err)
+	require.NotNil(t, ancestor)
+	require.Equal(t, root.hash, ancestor)
```

Comment:
I think these should be different tests.  I would use anonymous struct method to define tests.